src/array/product.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Perform a Cartesian product of arrays, combining all elements from the
+ * input arrays into all possible combinations.
+ *
+ * @see https://radashi.js.org/reference/array/product
+ * @example
+ * ```ts
+ * product([['red', 'blue'], ['big', 'small'], ['fast', 'slow']])
+ * // => [['red', 'big', 'fast'], ['red', 'big', 'slow'], ['red', 'small', 'fast'], ['red', 'small', 'slow'], ['blue', 'big', 'fast'], ['blue', 'big', 'slow'], ['blue', 'small', 'fast'], ['blue', 'small', 'slow']]
```

Comment:
This example is very long and difficult to read, better to divide it into lines.
```suggestion
 * // => [
 * //   ['red', 'big', 'nice'],
 * //   ['red', 'big', 'great'],
 * //   ['red', 'small', 'nice'],
 * //   ['red', 'small', 'great'],
 * //   ['blue', 'big', 'nice'],
 * //   ['blue', 'big', 'great'],
 * //   ['blue', 'small', 'nice'],
 * //   ['blue', 'small', 'great']
 * // ]
```